flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/PeriodicMaterializationManager.java

```diff
@@ -118,7 +118,7 @@ public void start() {
 
             LOG.info("Task {} starts periodic materialization", subtaskName);
 
-            scheduleNextMaterialization(periodicMaterializeDelay + initialDelay);
+            scheduleNextMaterialization(initialDelay);
```

Comment:
This change affects the tests which use very high `periodicMaterializationDelay` to effictively disable materialization. There, `periodicMaterializationDelay` set to `Long.MAX` may or may not disable the materialization.

So we might want to (re)introduce a flag to disable materialization.

WDYT @curcur ?

Reply:
> So we might want to (re)introduce a flag to disable materialization.
 
Currently,  if `PERIODIC_MATERIALIZATION_INTERVAL` is negative, the periodic materialization will be disabled. Do you mean to introduce a **new** flag?

Reply:
I think we should have some way to disable the materialization (flag).

I am not a fan of using `periodicMaterializationDelay` very long to disable materialization. It is implicit and may lead error.

Reply:
I think disabling materialization for test is not the main purpose of this pr, right ?
(Diabling has been supported in [FLINK-26673](https://issues.apache.org/jira/browse/FLINK-26673))
About making the checkpoint deletion evenly, It is not enough for users to using changlog out of box without configuring other configs like [jobmanager.io-pool.size](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/config/#jobmanager-io-pool-size) ?
We may still need other strategy to solve the problem of the checkpoint deletion delay ?

Reply:
So now we have `periodicMaterializationDelay`, we have `initialDelay` and we have `periodicMaterializationDelay < 0` to indicate materialization disabled if I understand correctly?

Isn't this so confusing and misleading to the users?  What about `initialDelay` being very long or negative?

What is the problem to introduce a flag to indicate enable/disable? I think the original argument is that we do not want to introduce too many configs, but I think the current way is much more confusing and problematic.

Reply:
Sorry for the confusion, a negative value will prevent any materialization indeed.
I think both approaches have their both pros and cons (a separate flag is more explicit, but the user has to keep the two variables consistent).
But keeping negative value as it is doesn't change the API, so I'd chose this option.